### PR TITLE
MAYA-112227 - UV set renamed to st, st1, st2

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
Implements automatic UV set renaming.
- The sets will be renamed st, st1, st2, st3,... in the order they are found in the mesh.
- The original name will be stored in custom data for roundtripping
- On import, the names will always be restored if the roundtripping information is found
- On import, the UV sets are restored in the same order as found in the USD primitive
- The environment variables MAYAUSD_IMPORT_PRIMARY_UV_SET_AS_MAP1 and  MAYAUSD_EXPORT_MAP1_AS_PRIMARY_UV_SET were removed as their functionnality is subsumed in the global renaming.

Caveat:
If you have a mesh with multiple UV sets, and a UV set named "map1" is not the first UV set, then it will not automatically be renamed "st" as was previously the case using MAYAUSD_EXPORT_MAP1_AS_PRIMARY_UV_SET. We expect these cases to be quite rare, but if you have post-processing scripts that are expecting "st" and scenes where "map1" is not the first UV sets; then the results at the other end of the post-processing pipeline will be different.

### CODE REVIEW

File: test/lib/usd/translators/UsdExportUVSetMappingsTest/UsdExportUVSetMappingsTest.ma [hunk 1]
```diff
@@ -148,7 +173,7 @@ createNode transform -n "pPlane3";
 	rename -uid "A9516DAE-46FB-C555-7036-F5BFA863699B";
 	setAttr ".t" -type "double3" 2.4 0 0 ;
 	setAttr ".r" -type "double3" 89.999999999999986 0 0 ;
-createNode mesh -n "pPlaneShape3" -p "pPlane3";
+createNode mesh -n "pPlane3" -p "|pPlane3";
```
Comment (JGamache-autodesk) [inline test/lib/usd/translators/UsdExportUVSetMappingsTest/UsdExportUVSetMappingsTest.ma:176]: @dgovil , @ysiewappl This makes the shape name identical to the transform name and makes sure the code in _UVMappingManager is still correctly handling that situation.

File: test/lib/usd/translators/testUsdExportUVSetMappings.py [hunk 2]
```diff
@@ -67,24 +64,15 @@ def testExportUVSetMappings(self):
...
         '''
         expected = [
-            ("/pPlane1", "/blinn1SG_map1_map1_map1", "map1", "map1", "map1"),
-            ("/pPlane2", "/blinn1SG", "st1", "st1", "st1"),
-            ("/pPlane3", "/blinn1SG", "st1", "st1", "st1"),
-            ("/pPlane4", "/blinn1SG_st2_st2_st2", "st2", "st2", "st2"),
-            ("/pPlane5", "/blinn1SG_p5a_p5b_p5c", "p5a", "p5b", "p5c"),
-            ("/pPlane6", "/blinn1SG_p62_p63_p61", "p62", "p63", "p61"),
-            ("/pPlane7", "/blinn1SG_p7r_p7p_p7q", "p7r", "p7p", "p7q"),
+            ("/pPlane1", "/blinn1SG", "st", "st", "st"),
```
Comment (JGamache-autodesk) [inline test/lib/usd/translators/testUsdExportUVSetMappings.py:67]: The automatic renaming does reduce the number of specializations created.

**Comment by kxl-adsk** [comment]
@dj-mcg Can you help review this change and confirm that it works for Pixar as well?

**Review by kxl-adsk — APPROVED** [review]
LGTM!

**Comment by dj-mcg** [comment]
This looks good to me, and I believe it allows me to delete one of our internal patches. I'll fix up whatever internal tests are affected, so lgtm!

**Review by dj-mcg — APPROVED** [review]

**Pull request merged**